server/impl/nonce/crypto.go

```diff
@@ -0,0 +1,57 @@
+package nonce
+
+import (
+	"bufio"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"io"
+	mathrand "math/rand"
+	"sync"
+)
+
+const cryptoRandBufferSize = 128
+
+// cryptoRandSource is an implementation of rand.Source64 backed by crypto/rand.
+type cryptoRandSource struct {
+	r   *bufio.Reader
+	buf []byte
+}
+
+var (
+	_ mathrand.Source   = cryptoRandSource{}
+	_ mathrand.Source64 = cryptoRandSource{}
```

Comment:
This asserts that cryptoRandSource implements both math/rand.Source and math/rand.Source64.